onchaind/onchain.c

```diff
@@ -243,10 +243,11 @@ static struct bitcoin_tx *tx_to_us(const tal_t *ctx,
 			 + 1 + 3 + 73 + 0 + tal_len(wscript))
 		/ 1000;
 
-	/* Result is trivial?  Just eliminate output. */
-	if (tx->output[0].amount < dust_limit_satoshis + fee)
-		tal_resize(&tx->output, 0);
-	else
+	/* Result is trivial?  Spent to OP_RETURN to avoid leaving dust. */
+	if (tx->output[0].amount < dust_limit_satoshis + fee) {
```

Comment:
This solves the to_us respend issue when the value is below the fee I guess. Wouldn't it be better to not create that to_us output at all? But that'd require signaling to the peer that we're ok omitting our output in the commit, which is a spec change.

Reply:
This is onchain-only: if fees rise and we can't spent, it's best to remove the UTXO.

The alternative would be to drop the feerate and cross our fingers (or consider it closed?).

For some outputs we're not in a hurry, it's just tidying up: they could potentially use a lower fee rate.